libraries/standard/mqtt/cbmc/sources/mqtt_cbmc_state.c

```diff
@@ -199,3 +207,64 @@ bool isValidMqttSubscriptionList( MQTTSubscribeInfo_t * pSubscriptionList,
 
     return isValid;
 }
+
+MQTTContext_t * allocateMqttContext( MQTTContext_t * pContext )
+{
+    TransportInterface_t * pTransportInterface;
+    MQTTApplicationCallbacks_t * pCallbacks;
+    MQTTFixedBuffer_t * pNetworkBuffer;
+    MQTTStatus_t status = MQTTSuccess;
+
+    if( pContext == NULL )
+    {
+        pContext = mallocCanFail( sizeof( MQTTContext_t ) );
+    }
+
+    pTransportInterface = mallocCanFail( sizeof( TransportInterface_t ) );
+
+    if( pTransportInterface != NULL )
+    {
+        pTransportInterface->recv = NetworkInterfaceReceiveStub;
+        pTransportInterface->send = NetworkInterfaceSendStub;
+    }
+
+    pCallbacks = mallocCanFail( sizeof( MQTTApplicationCallbacks_t ) );
+
+    if( pCallbacks )
+    {
+        pCallbacks->getTime = GetCurrentTimeStub;
+        pCallbacks->appCallback = EventCallbackStub;
+    }
+
+    pNetworkBuffer = allocateMqttFixedBuffer( NULL );
+    __CPROVER_assume( isValidMqttFixedBuffer( pNetworkBuffer ) );
+
+    /* It is part of the API contract to call MQTT_Init() with the MQTTContext_t
+     * before any other function in mqtt.h. */
+    if( pContext != NULL )
+    {
+        status = MQTT_Init( pContext, pTransportInterface, pCallbacks, pNetworkBuffer );
+    }
+
+    /* If the MQTTContext_t initialization failed, then set the context to NULL
+     * so that function under harness will return immediately upon a NULL
+     * parameter check. */
+    if( status != MQTTSuccess )
+    {
+        pContext = NULL;
+    }
+
+    return pContext;
+}
+
+bool isValidMqttContext( const MQTTContext_t * pContext )
+{
+    bool isValid = true;
+
+    if( pContext != NULL )
+    {
+        isValid = pContext->networkBuffer.size < CBMC_MAX_OBJECT_SIZE;
+    }
+
```

Comment:
There is an awful lot of assumptions being made in the allocator itself (eg, what can be null) and it would be good form to test all of those assumptions here.  The idea is to make is easy to see all of the assumptions in one place.

Reply:
Sure thank you